src/sidebar/services/frame-sync.ts

```diff
@@ -267,6 +282,33 @@ export class FrameSyncService {
           this._updateAnchorStatus(anchorImmediately, 'anchored');
         }
 
+        // For added annotations that still need to anchor, re-check them after
+        // ANCHORING_TIMEOUT has elapsed. Any added annotations not anchored
+        // by that time should be marked as having an anchor timeout.
+        const anchoringIds = added
+          .filter(ann => isWaitingToAnchor(ann))
```

Comment:
Annotations are immutable so any that were marked for immediate anchoring above will still be present in the `added` list. A possible solution would be to turn `anchorImmediately` into a set and filter against that.